### PR TITLE
Initial commit - headless v1.0

### PR DESCRIPTION
**Pull request by lukaszuznanski**

### CODE REVIEW

File: Classes/DataProcessing/ImageProcessor.php [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+        return $data;
+    }
+
+    /**
+     * @param FileReference $image
+     * @param array $dimensions
+     * @return ProcessedFile
+     */
+    protected function processImageFile(FileReference $image, array $dimensions): ProcessedFile
```
Comment (tmotyl) [inline Classes/DataProcessing/ImageProcessor.php]: is this code taken somewhere from TYPO3 core?

File: Classes/DataProcessing/MenuProcessor.php [hunk 2]
```diff
@@ -0,0 +1,542 @@
...
+ * includeSpacer = If true, pagetype spacer will be included in the menu
+ * titleField = Field that should be used for the title
+ *
+ * See HMENU docs for more options.
+ * https://docs.typo3.org/typo3cms/TyposcriptReference/ContentObjects/Hmenu/Index.html
+ *
+ *
+ * Example TypoScript configuration:
+ *
+ * 10 = TYPO3\CMS\Frontend\DataProcessing\MenuProcessor
```
Comment (tmotyl) [inline Classes/DataProcessing/MenuProcessor.php]: this example is wrong?
wouldn't it make sense to inherit from TYPO3\CMS\Frontend\DataProcessing\MenuProcessor ?

File: Classes/ContentObject/JsonContentObject.php [hunk 3]
```diff
@@ -0,0 +1,211 @@
+<?php
+namespace FriendsOfTYPO3\Headless\ContentObject;
```
Comment (tmotyl) [inline Classes/ContentObject/JsonContentObject.php]: missing strict type

File: Classes/DataProcessing/MenuProcessor.php [hunk 4]
```diff
@@ -0,0 +1,542 @@
+<?php
+namespace FriendsOfTYPO3\Headless\DataProcessing;
```
Comment (tmotyl) [inline Classes/DataProcessing/MenuProcessor.php]: strict type

File: Classes/Hooks/TypolinkHook.php [hunk 5]
```diff
@@ -0,0 +1,50 @@
+<?php
+declare(strict_types = 1);
+
+namespace FriendsOfTYPO3\Headless\Hooks;
+
+use TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer;
+
+/***
```
Comment (tmotyl) [inline Classes/Hooks/TypolinkHook.php]: hmm, let's unify license headers please

File: Classes/Utility/ContentUtility.php [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ *
+ *  (c) 2019
+ *
+ ***/
+
+/**
+ * ContentUtility
```
Comment (tmotyl) [inline Classes/Utility/ContentUtility.php]: what does it do?

File: Classes/Utility/ContentUtility.php [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+ */
+class ContentUtility
+{
+    /**
+     * @param $content
```
Comment (tmotyl) [inline Classes/Utility/ContentUtility.php]: plz add description

File: README.md [hunk 8]
```diff
@@ -1,2 +1,87 @@
...
+      }
+}
+```
+Output
+```
+{
+    "header" : "headerFieldValue"
+}
+```
+As you can see, we use reserved keyword `fields` to render output. Keys, which are inside `fields` are used as keys in json, so if you specify key `abc = COA`, then in json output you will have
```
Comment (tmotyl) [inline README.md]: abc= COA ? I don't get this example. Lets remove it and use the next one only

File: composer.json [hunk 9]
```diff
@@ -0,0 +1,37 @@
+{
+  "name": "friendsoftypo3/headless",
+  "type": "typo3-cms-extension",
+  "version": "1.0.0-beta",
```
Comment (tmotyl) [inline composer.json]: remove version form here, we will tag the release on github and register it in packagist

File: composer.json [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+      "name": "Łukasz Uznański",
+      "role": "Developer"
+    },
+    {
+      "name": "headless Development Team",
+      "role": "Developer"
+    }
+  ],
+  "require": {
+    "typo3/cms-core": "~9.5"
```
Comment (tmotyl) [inline composer.json]: "typo3/cms-core": "^9.5 || ^10.0",

File: Classes/DataProcessing/ImageProcessor.php (hunk 1 above)
Comment (lukaszuznanski) [reply Classes/DataProcessing/ImageProcessor.php]: yes, from ImageViewHelper

File: composer.json (hunk 10 above)
Comment (lukaszuznanski) [reply composer.json]: done

File: composer.json (hunk 9 above)
Comment (lukaszuznanski) [reply composer.json]: done

File: README.md (hunk 8 above)
Comment (lukaszuznanski) [reply README.md]: done

File: Classes/Utility/ContentUtility.php (hunk 7 above)
Comment (lukaszuznanski) [reply Classes/Utility/ContentUtility.php]: All methods in this class now have description

File: Classes/Utility/ContentUtility.php (hunk 6 above)
Comment (lukaszuznanski) [reply Classes/Utility/ContentUtility.php]: Added description

File: Classes/Hooks/TypolinkHook.php (hunk 5 above)
Comment (lukaszuznanski) [reply Classes/Hooks/TypolinkHook.php]: All license headers are unified now

File: Classes/DataProcessing/MenuProcessor.php (hunk 4 above)
Comment (lukaszuznanski) [reply Classes/DataProcessing/MenuProcessor.php]: done

File: Classes/ContentObject/JsonContentObject.php (hunk 3 above)
Comment (lukaszuznanski) [reply Classes/ContentObject/JsonContentObject.php]: done

File: Classes/DataProcessing/MenuProcessor.php (hunk 2 above)
Comment (lukaszuznanski) [reply Classes/DataProcessing/MenuProcessor.php]: That's true, now we inherit from core MenuProcessor and only override one property, example also fixed.

**Pull request merged**